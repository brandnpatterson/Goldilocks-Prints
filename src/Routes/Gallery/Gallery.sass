@import ../../index.sass

.gallery
  .landing
    @extend .centered-parent
    @extend .text-shadow
    background: none
    color: $white
    height: 300px
    margin: 20px 0
    padding: 50px 0
    width: 100vw
    h2
      @keyframes slide-down
        0%
          top: 230px
          opacity: 0
        100%
          top: 440px
          opacity: 1
      animation: slide-down 1s
      animation-fill-mode: forwards
      backface-visibility: hidden
      perspective: 1000
      font-size: 40px
      letter-spacing: 10px
      +mediaQuery($small)
        font-size: 30px
        letter-spacing: 5px
      +mediaQuery($smallest)
        font-size: 30px
        letter-spacing: 5px
      text-align: center
      position: absolute
      z-index: -10
