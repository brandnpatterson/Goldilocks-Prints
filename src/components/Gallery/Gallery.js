import React, { Component } from 'react'
import { Link } from 'react-router-dom'

class Gallery extends Component {
  render () {
    const location = window.location.pathname
    const art = location.match(/^\/art/) ? 'active' : ''
    const photography = location.match(/^\/photography/) ? 'active' : ''
    const design = location.match(/^\/design/) ? 'active' : ''

    return (
      <div className='gallery'>
        <div className='flex-center'>
          <h2>GALLERY</h2>
          <ul className='flex-center'>
            <li><Link id={art} to='/art'>ART</Link></li>
            <li><Link id={photography} to='/photography'>PHOTOGRAPHY</Link></li>
            <li><Link id={design} to='/design'>DESIGN</Link></li>
          </ul>
        </div>
      </div>
    )
  }
}

export default Gallery
