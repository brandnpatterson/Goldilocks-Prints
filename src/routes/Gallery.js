import React from 'react'

const Gallery = () => {
  return (
    <div className='gallery'>
      <div className='landing'>
        <h2>Pictures & Art For Sale</h2>
      </div>
      <div className='panel'>
        <h2>Gallery</h2>
      </div>
    </div>
  )
}

export default Gallery
