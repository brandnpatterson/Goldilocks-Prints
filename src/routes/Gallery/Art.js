import React from 'react'

const Art = () => {
  return (
    <div className='art'>
      <div className='panel'>
        <h2>Art</h2>
      </div>
    </div>
  )
}

export default Art
