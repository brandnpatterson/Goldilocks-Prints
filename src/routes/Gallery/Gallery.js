import React from 'react'
import { Link } from 'react-router-dom'

const Gallery = () => {
  return (
    <div className='gallery'>
      <div className='panel'>
        <h2>GALLERY</h2>
        <ul className='flex-container'>
          <li><Link to='/art'>ART</Link></li>
          <li><Link to='/photography'>PHOTOGRAPHY</Link></li>
          <li><Link to='/design'>DESIGN</Link></li>
        </ul>
      </div>
    </div>
  )
}

export default Gallery
