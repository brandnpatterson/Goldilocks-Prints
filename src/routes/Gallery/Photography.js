import React from 'react'

const Photography = () => {
  return (
    <div className='photography'>
      <div className='panel'>
        <img src='../../public/images/gallery/senior-pics1.jpg' alt='Senior Pics' />
        <img src='../../public/images/gallery/senior-pics2.jpg' alt='Senior Pics' />
        <img src='../../public/images/gallery/senior-pics3.jpg' alt='Senior Pics' />
        <img src='../../public/images/gallery/senior-pics4.jpg' alt='Senior Pics' />
        <img src='../../public/images/gallery/senior-pics5.jpg' alt='Senior Pics' />
        <img src='../../public/images/gallery/senior-pics6.jpg' alt='Senior Pics' />
        <img src='../../public/images/gallery/senior-pics7.jpg' alt='Senior Pics' />
        <img src='../../public/images/gallery/senior-pics8.jpg' alt='Senior Pics' />
      </div>
    </div>
  )
}

export default Photography
