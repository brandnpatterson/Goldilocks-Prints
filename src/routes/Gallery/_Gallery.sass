.gallery
  padding-top: 100px
  width: 100vw
  ul
    padding-top: 40px
  h2,
  li
    color: $white
    display: inline-block
    max-width: 70%
    letter-spacing: 5px
  h2
    font-family: $mainFont
  .flex-center
    flex-direction: row
    +mediaQuery($small)
      justify-content: space-around
    padding-left: 0
    a
      border: 2px $black solid
      padding: 15px 5px
      font-size: 10px
      +mediaQuery($small)
        padding: 20px
        font-size: 15px
      color: $white
.art,
.photography,
.design
  background: $black
  .image-panel
    background: $black
    flex-direction: row
    max-width: 100%
  .panel-gray
    width: 100vw
    height: 70px
    margin: 50px auto
    h2
      color: $black
      letter-spacing: 5px
      padding-top: 5px
  h2,
  h3,
  li
    color: $white
.photography
  .image-panel
    img
      +mediaQuery($small)
        width: 400px
.photography
  .image-panel.large
    img
      +mediaQuery($small)
        width: 200px
#active
  border: 2px $white solid
