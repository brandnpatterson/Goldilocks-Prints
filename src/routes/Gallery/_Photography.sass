.art,
.photography,
.design
  background: $black
  .image-panel
    background: $black
    flex-direction: row
    max-width: 100%
  .panel-gray
    width: 100vw
    height: 70px
    margin: 50px auto
    h2
      color: $black
      letter-spacing: 5px
      padding-top: 5px
  h2,
  h3,
  li
    color: $white
.photography
  .image-panel
    img
      +mediaQuery($small)
        width: 400px
.photography
  .image-panel.large
    img
      +mediaQuery($small)
        width: 200px
#active
  border: 2px $white solid
