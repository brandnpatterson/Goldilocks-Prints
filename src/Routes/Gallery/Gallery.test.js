import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Gallery from './Gallery'

test('It should render the Gallery component as expected', () => {
  const component = shallow(<Gallery />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
