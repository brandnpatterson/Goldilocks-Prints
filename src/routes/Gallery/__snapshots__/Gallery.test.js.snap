// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the Gallery component as expected 1`] = `
<div
  className="gallery"
>
  <div
    className="landing"
  >
    <h2>
      Pictures & Art For Sale
    </h2>
  </div>
  <div
    className="panel"
  >
    <h2>
      Gallery
    </h2>
  </div>
</div>
`;
